include/XLink/XLinkMacros.h

```diff
@@ -6,17 +6,14 @@
 #define MVMACROS_H__
 
 #define CIRCULAR_INCREMENT(x, maxVal) \
-    { \
-         x++; \
-         if (x == maxVal) \
-             x = 0; \
+    {                                 \
+        x = (++x) % maxVal;           \
```

Comment:
This results in `warning: multiple unsequenced modifications to`.
Could be `(x+1) % maxVal` to not trigger sequence point warnings, or kept as before?
@diablodale

Reply:
How can I reproduce such an error? platform, os, compile switches, etc.
Are you compiling with something that isn't guaranteed c++11? (though it should be guaranteed c++14)

As I intended this...
1. explicit paranthesis to force specific sequencing.
2. x is `prefix` incremented by 1, prefix operates directly on memory, returns a ref to that memory after it has been incremented. No new temp var and copy is made. https://en.cppreference.com/w/cpp/language/operator_incdec
3. ref to x is mod by maxVal
4. Results of that mod is saved into x

My goal in the change is to prevent create and copy of additional temporaries -and- to remove the if branch. Branching is bad in tight loops like this circular buffer. This code needs to be computationally efficient. Haha, correct too 😉

Reply:
I can't reproduce your warning on Ubuntu 20.04 with gcc 10.3. Still, I respect its caution.
Reading the C spec, perhaps their is an unclear seq point on your specific setup.
It might be resolved by updating the code with a little of the new and a little of the old.
The following works equally well on my Windows build and all test+examples pass.

```c++
#define CIRCULAR_INCREMENT(x, maxVal) \
    {                                 \
        ++x;                          \
        x %= maxVal;                  \
    }

#define CIRCULAR_INCREMENT_BASE(x, maxVal, base) \
    {                                            \
        ++x;                                     \
        if (x == maxVal)                         \
            x = base;                            \
    }

```

Reply:
Ubuntu 21.10, `clang 13.0.0-2`.

Reply:
still no repro. 🤷‍♀️
Ubuntu 21.10 with
Ubuntu clang version 13.0.1-++20220120093215+6349865a8f37-1~exp1~20220120213226.62
Target: x86_64-pc-linux-gnu
Thread model: posix

depthai-core 7651e7627fa83e76848fdb7bff23b2636011e262
xlink local with 8a2ea2741b70dc5179952ad52aa80e126240e852
build debug and shared_libs
Here is the entire cmake config and build log
[xlink-macro.log](https://github.com/luxonis/XLink/files/7910113/xlink-macro.log)

The code change I have above is logically the same but may be more specific for your setup.
I am not able test if it makes "better" since I do not get that warning on U20 gcc, U21 clang, or Win vs2019.

I will start a static build to see if U21 clang behaves differently.

Reply:
No repro on dynamic, static, debug, or release.

Reply:
Can you make a dockerfile that reproduces the issue? U can then send that to me and I can debug with that.

Reply:
Sorry, reproducible on `mdk_r17` branch: https://github.com/luxonis/XLink/pull/21
Just build XLink locally: `mkdir build; make ..; make -j`

Reply:
If you are still interested you can try commit ed9970a02cbc88724623c450daf4d1d2f6ee5e44

I just brought up to understand why it was changed. The change to suppress it was trivial as suggested.

Reply:
It is good review feedback. Thank you! :-)
By C-spec, I think my code was correctly identified as ambiguous.
Two things were in play..i think...it is so very subtle.
1. spec for a prefix incrementer is different between C and C++.. Its a subtle difference in returning value vs. returning reference. My head was in "C++" thinking when I removed the if branch in the macro and I did not consider that difference.
2. C has a spec rule that inbetween two sequence points, a "variable" can be modified only once. My code that generated that warning, by C spec, modified it twice between sequence points. The `++` causes an modification by a C "sideaffect" and then the `=` assign caused a second.

Reply:
We could keep the "same form" with little modification, by using the underrated comma operator, which is a sequence point.
`x = (++x, x % maxVal);`
It's bad code though, and shouldn't be used over `(x+1) % maxVal`.

Warning was given only for:
```
uint32_t packetId = stream->firstPacket;
CIRCULAR_INCREMENT(packetId, XLINK_MAX_PACKETS_PER_STREAM);
```
and not for:
```
streamDesc_t* stream;
CIRCULAR_INCREMENT(stream->firstPacket, XLINK_MAX_PACKETS_PER_STREAM);
```

Which is interesting.

Reply:
Awww, poor comma. 😭😂 I only use it to hack text into a windows assert `assert(("my text", test == value));`